src/ctia/stores/es/crud.clj

```diff
@@ -202,24 +213,125 @@ It returns the documents with full hits meta data including the real index in wh
   [docs ident get-in-config]
   (filter #(allow-read? % ident get-in-config) docs))
 
+(s/defschema BulkResult
+  (st/optional-keys
+   {:deleted [s/Str]
+    :updated [s/Str]
+    :errors (st/optional-keys
+             {:forbidden [s/Str]
+              :not-found [s/Str]
+              :internal-error [s/Str]})}))
+
+(defn- format-bulk-res
+  [bulk-res]
+  (let [{:keys [deleted updated not_found] :as res}
+        (->> (:items bulk-res)
+             (map (comp first vals))
+             (group-by :result)
+             (into {}
+                   (map (fn [[result items]]
+                          {(keyword result) (map :_id items)}))))]
+    (cond-> {}
+      deleted (assoc :deleted deleted)
+      updated (assoc :updated updated)
+      not_found (assoc-in [:errors :not-found] not_found))))
+
+(defn check-and-prepare-bulk
+  [{{{:keys [get-in-config]} :ConfigService} :services
+    :as conn-state}
+   ids
+   ident]
+  (let [doc-ids (map ensure-document-id ids)
+        docs-with-indices (get-docs-with-indices conn-state doc-ids {})
```

Comment:
generic remark, but it would be very useful to find a version of `get-docs-with-indices` that could take a ident as parameter to build an ES request that will returns only docs the ident can read (or write depending on a param).